x-pack/plugin/ilm/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/ilm/TimeSeriesLifecycleActionsIT.java

```diff
@@ -223,6 +228,40 @@ public void testRolloverActionWithIndexingComplete() throws Exception {
             getOnlyIndexSettings(client(), originalIndex).get(LifecycleSettings.LIFECYCLE_INDEXING_COMPLETE)));
     }
 
+    public void testRolloverActionWithMaxSinglePrimarySize() throws Exception {
+        String originalIndex = index + "-000001";
+        String secondIndex = index + "-000002";
+        createIndexWithSettings(client(), originalIndex, alias, Settings.builder()
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 3)
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+            .put(RolloverAction.LIFECYCLE_ROLLOVER_ALIAS, alias));
+
+        int totalDocs = randomIntBetween(16, 32);
+        for (int i = 0; i < totalDocs; i++) {
+            index(client(), originalIndex, String.valueOf(i), "foo", "bar_" + i);
+        }
+        refresh(client(), originalIndex);
+
+        List<Integer> primaryShardSizes = getPrimaryShardSizes(client(), originalIndex);
+        Integer largest = Collections.max(primaryShardSizes);
+
+        // create policy
+        createNewSingletonPolicy(client(), policy, "hot", new RolloverAction(
+            null,
+            // the largest size we have from above is index stats size, but that's larger
+            // than the docstats size that we actually evaluate against inside the guts
+            // of rollover, 2048 is a slop factor
+            ByteSizeValue.ofBytes(Long.max(1, largest - 2048)),
```

Comment:
I am inclined to say this sort of complexity is not needed here. We're not meant to test the accuracy of shard size stats here, but that the condition is triggered. 
Shall we just use `1` here and also delete `getPrimaryShardSizes`? 

What do you think?

Reply:
4393939